app/components/autocomplete/slash_suggestion/app_slash_suggestion/app_slash_suggestion.tsx

```diff
@@ -0,0 +1,238 @@
+        this.setState({active});
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (!isMinimumServerVersion(Client4.getServerVersion(), 5, 24) || !this.props.appsEnabled) {
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        if (
+            (this.props.value === prevProps.value) ||
+            this.props.isSearch || this.props.value.startsWith('//') || !this.props.channelId
+        ) {
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        const {
+            value: nextValue,
+        } = this.props;
+
+        if (nextValue[0] !== '/') {
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        if (nextValue.indexOf(' ') === -1) { // delegate base command to command parser
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        if (!this.isAppCommand(nextValue, this.props.channelId, this.props.currentTeamId, this.props.rootId)) {
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        this.fetchAndShowAppCommandSuggestions(nextValue, this.props.channelId, this.props.currentTeamId, this.props.rootId);
+    }
+
+    isAppCommand = (pretext: string, channelID: string, teamID = '', rootID?: string) => {
+        this.appCommandParser.setChannelContext(channelID, teamID, rootID);
+        return this.appCommandParser.isAppCommand(pretext);
+    }
+
+    fetchAndShowAppCommandSuggestions = async (pretext: string, channelID: string, teamID = '', rootID?: string) => {
+        this.appCommandParser.setChannelContext(channelID, teamID, rootID);
+        const suggestions = await this.appCommandParser.getSuggestions(pretext);
+        this.updateSuggestions(suggestions);
+    }
+
+    updateSuggestions = (matches: ExtendedAutocompleteSuggestion[]) => {
+        this.setState({
+            active: Boolean(matches.length),
+            dataSource: matches,
+        });
+        this.props.onResultCountChange(matches.length);
+    }
+
+    completeSuggestion = (command: string) => {
+        const {onChangeText} = this.props;
+        analytics.trackCommand('complete_suggestion', `/${command} `);
+
+        // We are going to set a double / on iOS to prevent the auto correct from taking over and replacing it
+        // with the wrong value, this is a hack but I could not found another way to solve it
+        let completedDraft = `/${command} `;
+        if (Platform.OS === 'ios') {
+            completedDraft = `//${command} `;
+        }
+
+        onChangeText(completedDraft);
+
+        if (Platform.OS === 'ios') {
+            // This is the second part of the hack were we replace the double / with just one
+            // after the auto correct vanished
+            setTimeout(() => {
```

Comment:
make sure you clear the setTimeouts to avoid memory leaks

Reply:
Not sure how necessary it is to clear them, since they are at 0ms.